scripts/code_size_compare.py

```diff
@@ -51,15 +102,61 @@ def __init__(self, old_revision, new_revision, result_dir):
 
         self.old_rev = old_revision
         self.new_rev = new_revision
+        self.arch = arch
+        self.config = config
         self.git_command = "git"
-        self.make_command = "make"
+        self.pre_build_commands = PRE_BUILD_CMDS[config]
+        self.make_command = ''
+        self._set_make_command()
+
+        self.old_sizes = {}
+        self.new_sizes = {}
+        self.change_pcts = {}
 
     @staticmethod
     def validate_revision(revision):
         result = subprocess.check_output(["git", "rev-parse", "--verify",
                                           revision + "^{commit}"], shell=False)
         return result
 
+    def _set_make_command(self):
+        """Use the config and arch options passed to the object to determine
+           and set the correct build command."""
+        if self.arch == 'x86' and (self.config in {'default', 'full',\
+                                                   'baremetal'}):
+            self.make_command = 'make -j lib'
+            return
+
+        # Default just takes the current config, which may or may not work
+        # with baremetal targets. Warn the user.
+        if self.config == 'default':
+            print("Assuming that the current config is compatible with \
+                   baremetal targets. If it isn't the build may fail!")
+
+        if self.config in {'default', 'baremetal'}:
+            if self.arch == 'aarch32':
+                self.make_command = 'make -j lib CC=armclang\
+                                    CFLAGS=\"--target=arm-arm-none-eabi \
+                                    -mcpu=cortex-m33 -Os\"'
+            if self.arch == 'aarch64':
```

Comment:
I'm still seeing build errors regarding `aarch32` and `aarch64`. Do I miss something?
```
Assuming that the current config is compatible with                    baremetal targets. If it isn't the build may fail!
Creating git worktree for c2cd8cd846f4ac49188db6033d3336920a876f27
The following command: make -j lib CC=/usr/local/ArmCompilerforEmbedded6.19/bin/armclang                                    CFLAGS="--target=aarch64-arm-none-eabi" failed and exited with code 2
Process output:
 b'  CC    aes.c\n  CC    aesni.c\n  CC    aesce.c\n  CC    aria.c\n  CC    asn1parse.c\n  CC    asn1write.c\n  CC    base64.c\n  CC    bignum.c\n  CC    bignum_core.c\n  CC    bignum_mod.c\n  CC    bignum_mod_raw.c\n  CC    camellia.c\n  CC    ccm.c\n  CC    chacha20.c\n  CC    chachapoly.c\n  CC    cipher.c\n  CC    cipher_wrap.c\n  CC    cmac.c\n  CC    constant_time.c\n  CC    ctr_drbg.c\n  CC    des.c\n  CC    dhm.c\n  CC    ecdh.c\n  CC    ecdsa.c\n  CC    ecp.c\n  CC    md.c\n  CC    lmots.c\n  CC    hmac_drbg.c\n  CC    entropy.c\n  CC    gcm.c\n  CC    entropy_poll.c\n  Gen   error.c\n  CC    md5.c\n  CC    psa_util.c\n  CC    nist_kw.c\n  CC    platform.c\n  CC    psa_crypto.c\n  CC    pk.c\n  CC    psa_crypto_aead.c\n  CC    pkwrite.c\n  CC    pkcs12.c\n  CC    pkparse.c\n  CC    ecp_curves.c\n  CC    oid.c\n  CC    hkdf.c\n  CC    pk_wrap.c\n  CC    ecjpake.c\n  CC    padlock.c\n  CC    pem.c\n  CC    psa_crypto_se.c\n  CC    pkcs5.c\n  CC    psa_crypto_cipher.c\n  CC    x509.c\n  CC    psa_crypto_rsa.c\n  CC    psa_its_file.c\n  CC    psa_crypto_storage.c\n  CC    psa_crypto_slot_management.c\n  CC    sha1.c\n  Gen   psa_crypto_driver_wrappers.c\n  CC    psa_crypto_mac.c\n  CC    ripemd160.c\n  CC    sha256.c\n  CC    timing.c\n  CC    psa_crypto_client.c\n  CC    ../3rdparty/everest/library/everest.c\n  CC    hash_info.c\n  CC    psa_crypto_ecp.c\n  CC    poly1305.c\n  CC    rsa.c\n  CC    memory_buffer_alloc.c\n  CC    psa_crypto_pake.c\n  CC    x509_csr.c\n  CC    psa_crypto_hash.c\n  CC    x509_crl.c\n  CC    ../3rdparty/everest/library/Hacl_Curve25519_joined.c\n  CC    lms.c\n  CC    x509_create.c\n  CC    x509_crt.c\n  Gen   version_features.c\n  CC    ../3rdparty/everest/library/x25519.c\n  CC    rsa_alt_helpers.c\n  CC    platform_util.c\n  CC    threading.c\n  CC    x509write_crt.c\n  CC    sha512.c\n  CC    version.c\n  CC    pkcs7.c\n  CC    mps_trace.c\n  CC    debug.c\n  CC    x509write_csr.c\n  CC    ssl_ciphersuites.c\n  CC    ssl_cookie.c\n  CC    mps_reader.c\n  CC    ssl_client.c\n  CC    ssl_cache.c\n  CC    net_sockets.c\nentropy_poll.c:45:2: error: "Platform entropy sources only work on Unix and Windows, see MBEDTLS_NO_PLATFORM_ENTROPY in mbedtls_config.h"\n#error \\\n ^\n  CC    ssl_tls13_keys.c\n  Gen   ssl_debug_helpers_generated.c\n  CC    ssl_msg.c\n  CC    ssl_tls13_server.c\n  CC    ssl_tls.c\n  CC    ssl_ticket.c\n  CC    ssl_tls12_client.c\n  CC    ssl_tls12_server.c\n1 error generated.\n  CC    ssl_tls13_generic.c\nMakefile:305: recipe for target \'entropy_poll.o\' failed\nmake[1]: *** [entropy_poll.o] Error 1\nmake[1]: *** Waiting for unfinished jobs....\n  CC    ssl_tls13_client.c\ntiming.c:31:2: error: "This module only works on Unix and Windows, see MBEDTLS_TIMING_C in mbedtls_config.h"\n#error "This module only works on Unix and Windows, see MBEDTLS_TIMING_C in mbedtls_config.h"\n ^\ntiming.c:45:10: fatal error: \'unistd.h\' file not found\n#include <unistd.h>\n         ^~~~~~~~~~\n2 errors generated.\nMakefile:305: recipe for target \'timing.o\' failed\nmake[1]: *** [timing.o] Error 1\nnet_sockets.cplatform_util.c:261:2: error: "No mbedtls_ms_time available"\n#error "No mbedtls_ms_time available"\n ^\n:37:2: error: "This module only works on Unix and Windows, see MBEDTLS_NET_C in mbedtls_config.h"\n#error "This module only works on Unix and Windows, see MBEDTLS_NET_C in mbedtls_config.h"\n ^\n1 error generated.\nMakefile:305: recipe for target \'platform_util.o\' failed\nmake[1]: *** [platform_util.o] Error 1\npkcs7.c:30:10: fatal error: \'sys/types.h\' file not found\n#include <sys/types.h>\n         ^~~~~~~~~~~~~\n1 error generated.\nMakefile:305: recipe for target \'pkcs7.o\' failed\nmake[1]: *** [pkcs7.o] Error 1\nx509_crt.c:70:10: fatal error: \'sys/types.h\' file not found\n#include <sys/types.h>\n         ^~~~~~~~~~~~~\nnet_sockets.c:81:10: fatal error: \'sys/types.h\' file not found\n#include <sys/types.h>\n         ^~~~~~~~~~~~~\n2 errors generated.\nMakefile:305: recipe for target \'net_sockets.o\' failed\nmake[1]: *** [net_sockets.o] Error 1\n1 error generated.\nMakefile:305: recipe for target \'x509_crt.o\' failed\nmake[1]: *** [x509_crt.o] Error 1\nMakefile:18: recipe for target \'lib\' failed\nmake: *** [lib] Error 2\n'

```